library/network_connections.py

```diff
@@ -1098,8 +1111,15 @@ def connection_create(self, connections, idx, connection_current=None):
             for nameserver in ip["dns"]:
                 if nameserver["family"] == socket.AF_INET6:
                     s_ip6.add_dns(nameserver["address"])
-            for search_domain in ip["dns_search"]:
-                s_ip6.add_dns_search(search_domain)
+            # DNS (the protocol) is irrelevant to the IP family. NetworkManager allows
+            # you to confiugre `ipv[46].dns-*` options (separated by IP family), it is
+            # a misdesign in the API. The role implemented it correctly, but when these
+            # options need to be set, it should configure both IPv4 and IPv6 setting in
+            # NetworkManager. And NetworkManager disallows to configure
+            # `ipv6.dns-search` setting when `ipv6.method` is disabled or ignore.
+            if not is_ipv6_disabled and not is_ipv6_ignore:
```

Comment:
One boolean variable seems to be enough, you can call it for example ipv6_available if you want to avoid confusion with disabled or ignore.

Reply:
Nice hint ! Updated.